# Phylogenomics practical session

By Iker Isarri and M. Eleonora Rossi

## Introduction

In this repository you will find the data, the scripts and the instructions to run the tutorial of the phylogenomics session.



## What can you do with the content of this repository?

You can use the data and instructions provided in this repository to...

## Data

First we need to copy this directory on our computer
```sh
git clone https://github.com/iirisarri/PEB_Phylogenomics.git
```
or download it? (Zip file?)
Once unzipped we move inside the directory 
```sh 
cd [name directoy]
```
Let's have a look on what's inside by typing the command `ls` or `ll`

## Orthology inference



## EXTRA: Pre-alignment uality filtering

## Multiple sequence alignment

## EXTRA: Alignment trimming

## Concatenate the alignments

## Tree inference : Maximum likelihood

## EXTRA: Bayesian Inferene and Coalescence methods with ASTRAL

## Tree visualization

## Software links

* Orthofinder (https://github.com/davidemms/OrthoFinder)
* PREQUAL (https://github.com/simonwhelan/prequal)
* MAFFT (https://mafft.cbrc.jp/alignment/software/source.html)
* MUSCLE v5 (https://github.com/rcedgar/muscle)
* TrimAL (https://vicfero.github.io/trimal/)
* FASTCONCAT (https://github.com/PatrickKueck/FASconCAT-G)
* IQTREE2 (http://www.iqtree.org/)
* Phylobayes (https://github.com/bayesiancook/phylobayes/tree/master)
* ASTRAL (https://github.com/smirarab/ASTRAL)
* FIGTree V1.4.4 (https://github.com/rambaut/figtree/releases)
* iTOL (https://itol.embl.de/)
